### ISSUE TITLE
Disbursements endpoint

### ISSUE THREAD

**Issue by NinoSkopac** [PromisePay/promisepay-php#66]
Jean, please shed some light on this.

In June 2016, I was tasked with adding Disbursements endpoints to the SDK.

However, I don't see those endpoints in either pre-live or live reference.
Could you please explain?

The [tests](https://github.com/NinoSkopac/promisepay-php/blob/6c617b7c70996a9d21d14f078cf68f478eb26b2a/tests/DisbursementTest.php) work, so the endpoints definitely exist.

Dumping this
```php
$disbursements = PromisePay::Disbursement()->get();
```

returns

```
  [0]=>
  array(8) {
    ["id"]=>
    string(36) "d9fa5f7f-17f5-4b49-a61c-25305d63e6d0"
    ["amount"]=>
    int(100)
    ["currency"]=>
    string(3) "AUD"
    ["batch_id"]=>
    int(23984)
    ["created_at"]=>
    string(24) "2017-08-10T20:37:36.094Z"
    ["updated_at"]=>
    string(24) "2017-08-10T20:42:32.561Z"
    ["state"]=>
    string(10) "successful"
    ["links"]=>
    array(6) {
      ["transactions"]=>
      string(64) "/disbursements/d9fa5f7f-17f5-4b49-a61c-25305d63e6d0/transactions"
      ["wallet_accounts"]=>
      string(67) "/disbursements/d9fa5f7f-17f5-4b49-a61c-25305d63e6d0/wallet_accounts"
      ["bank_accounts"]=>
      string(65) "/disbursements/d9fa5f7f-17f5-4b49-a61c-25305d63e6d0/bank_accounts"
      ["paypal_accounts"]=>
      string(67) "/disbursements/d9fa5f7f-17f5-4b49-a61c-25305d63e6d0/paypal_accounts"
      ["items"]=>
      string(57) "/disbursements/d9fa5f7f-17f5-4b49-a61c-25305d63e6d0/items"
      ["users"]=>
      string(57) "/disbursements/d9fa5f7f-17f5-4b49-a61c-25305d63e6d0/users"
    }
  }
  [1]=>
  array(8) {
    ["id"]=>
    string(36) "a17012a7-2b83-405c-b954-6ae26c8af471"
    ["amount"]=>
    int(100)
    ["currency"]=>
    string(3) "AUD"
    ["batch_id"]=>
    int(23984)
    ["created_at"]=>
    string(24) "2017-08-10T20:37:43.451Z"
    ["updated_at"]=>
    string(24) "2017-08-10T20:42:33.823Z"
    ["state"]=>
    string(10) "successful"
    ["links"]=>
    array(6) {
      ["transactions"]=>
      string(64) "/disbursements/a17012a7-2b83-405c-b954-6ae26c8af471/transactions"
      ["wallet_accounts"]=>
      string(67) "/disbursements/a17012a7-2b83-405c-b954-6ae26c8af471/wallet_accounts"
      ["bank_accounts"]=>
      string(65) "/disbursements/a17012a7-2b83-405c-b954-6ae26c8af471/bank_accounts"
      ["paypal_accounts"]=>
      string(67) "/disbursements/a17012a7-2b83-405c-b954-6ae26c8af471/paypal_accounts"
      ["items"]=>
      string(57) "/disbursements/a17012a7-2b83-405c-b954-6ae26c8af471/items"
      ["users"]=>
      string(57) "/disbursements/a17012a7-2b83-405c-b954-6ae26c8af471/users"
    }
  }
  [2]=>
  array(13) {
    ["id"]=>
    string(36) "fe09769c-20bd-4048-a963-9b124a96b1a3"
    ["amount"]=>
    int(1000)
    ["currency"]=>
    string(3) "AUD"
    ["batch_id"]=>
    int(466)
    ["created_at"]=>
    string(24) "2016-07-20T05:39:41.730Z"
    ["updated_at"]=>
    string(24) "2016-07-20T06:01:13.021Z"
    ["state"]=>
    string(7) "batched"
    ["to"]=>
    string(12) "Bank Account"
    ["bank_name"]=>
    string(13) "bank for test"
    ["bank_account_name"]=>
    string(8) "test acc"
    ["bank_account_number"]=>
    string(6) "XXX134"
    ["bank_routing_number"]=>
    string(6) "XXXXX1"
    ["links"]=>
    array(6) {
      ["transactions"]=>
      string(64) "/disbursements/fe09769c-20bd-4048-a963-9b124a96b1a3/transactions"
      ["wallet_accounts"]=>
      string(67) "/disbursements/fe09769c-20bd-4048-a963-9b124a96b1a3/wallet_accounts"
      ["bank_accounts"]=>
      string(65) "/disbursements/fe09769c-20bd-4048-a963-9b124a96b1a3/bank_accounts"
      ["paypal_accounts"]=>
      string(67) "/disbursements/fe09769c-20bd-4048-a963-9b124a96b1a3/paypal_accounts"
      ["items"]=>
      string(57) "/disbursements/fe09769c-20bd-4048-a963-9b124a96b1a3/items"
      ["users"]=>
      string(57) "/disbursements/fe09769c-20bd-4048-a963-9b124a96b1a3/users"
    }
  }
  [3]=>
  array(8) {
    ["id"]=>
    string(36) "07aa6675-34d1-43ce-9fd7-67ac8588b3a0"
    ["amount"]=>
    int(100)
    ["currency"]=>
    string(3) "AUD"
    ["batch_id"]=>
    int(23984)
    ["created_at"]=>
    string(24) "2017-08-10T20:37:52.253Z"
    ["updated_at"]=>
    string(24) "2017-08-10T20:42:34.979Z"
    ["state"]=>
    string(10) "successful"
    ["links"]=>
    array(6) {
      ["transactions"]=>
      string(64) "/disbursements/07aa6675-34d1-43ce-9fd7-67ac8588b3a0/transactions"
      ["wallet_accounts"]=>
      string(67) "/disbursements/07aa6675-34d1-43ce-9fd7-67ac8588b3a0/wallet_accounts"
      ["bank_accounts"]=>
      string(65) "/disbursements/07aa6675-34d1-43ce-9fd7-67ac8588b3a0/bank_accounts"
      ["paypal_accounts"]=>
      string(67) "/disbursements/07aa6675-34d1-43ce-9fd7-67ac8588b3a0/paypal_accounts"
      ["items"]=>
      string(57) "/disbursements/07aa6675-34d1-43ce-9fd7-67ac8588b3a0/items"
      ["users"]=>
      string(57) "/disbursements/07aa6675-34d1-43ce-9fd7-67ac8588b3a0/users"
    }
  }
  [4]=>
  array(8) {
    ["id"]=>
    string(36) "31e3eb60-fbd6-46fd-a247-73414cc33b7a"
    ["amount"]=>
    int(100)
    ["currency"]=>
    string(3) "AUD"
    ["batch_id"]=>
    int(23984)
    ["created_at"]=>
    string(24) "2017-08-10T20:38:00.360Z"
    ["updated_at"]=>
    string(24) "2017-08-10T20:42:36.220Z"
    ["state"]=>
    string(10) "successful"
    ["links"]=>
    array(6) {
      ["transactions"]=>
      string(64) "/disbursements/31e3eb60-fbd6-46fd-a247-73414cc33b7a/transactions"
      ["wallet_accounts"]=>
      string(67) "/disbursements/31e3eb60-fbd6-46fd-a247-73414cc33b7a/wallet_accounts"
      ["bank_accounts"]=>
      string(65) "/disbursements/31e3eb60-fbd6-46fd-a247-73414cc33b7a/bank_accounts"
      ["paypal_accounts"]=>
      string(67) "/disbursements/31e3eb60-fbd6-46fd-a247-73414cc33b7a/paypal_accounts"
      ["items"]=>
      string(57) "/disbursements/31e3eb60-fbd6-46fd-a247-73414cc33b7a/items"
      ["users"]=>
      string(57) "/disbursements/31e3eb60-fbd6-46fd-a247-73414cc33b7a/users"
    }
  }
  [5]=>
  array(13) {
    ["id"]=>
    string(36) "658ee18c-1a06-4f9d-82e8-7ca6d9f18975"
    ["amount"]=>
    int(1000)
    ["currency"]=>
    string(3) "AUD"
    ["batch_id"]=>
    int(467)
    ["created_at"]=>
    string(24) "2016-07-20T06:29:51.095Z"
    ["updated_at"]=>
    string(24) "2016-07-20T22:01:26.191Z"
    ["state"]=>
    string(7) "batched"
    ["to"]=>
    string(12) "Bank Account"
    ["bank_name"]=>
    string(13) "bank for test"
    ["bank_account_name"]=>
    string(8) "test acc"
    ["bank_account_number"]=>
    string(6) "XXX134"
    ["bank_routing_number"]=>
    string(6) "XXXXX1"
    ["links"]=>
    array(6) {
      ["transactions"]=>
      string(64) "/disbursements/658ee18c-1a06-4f9d-82e8-7ca6d9f18975/transactions"
      ["wallet_accounts"]=>
      string(67) "/disbursements/658ee18c-1a06-4f9d-82e8-7ca6d9f18975/wallet_accounts"
      ["bank_accounts"]=>
      string(65) "/disbursements/658ee18c-1a06-4f9d-82e8-7ca6d9f18975/bank_accounts"
      ["paypal_accounts"]=>
      string(67) "/disbursements/658ee18c-1a06-4f9d-82e8-7ca6d9f18975/paypal_accounts"
      ["items"]=>
      string(57) "/disbursements/658ee18c-1a06-4f9d-82e8-7ca6d9f18975/items"
      ["users"]=>
      string(57) "/disbursements/658ee18c-1a06-4f9d-82e8-7ca6d9f18975/users"
    }
  }
  [6]=>
  array(13) {
    ["id"]=>
    string(36) "82544f72-fe75-409c-beee-77f4f23b3dda"
    ["amount"]=>
    int(1000)
    ["currency"]=>
    string(3) "AUD"
    ["batch_id"]=>
    int(479)
    ["created_at"]=>
    string(24) "2016-07-26T21:43:32.215Z"
    ["updated_at"]=>
    string(24) "2016-07-26T22:01:07.016Z"
    ["state"]=>
    string(7) "batched"
    ["to"]=>
    string(12) "Bank Account"
    ["bank_name"]=>
    string(13) "bank for test"
    ["bank_account_name"]=>
    string(8) "test acc"
    ["bank_account_number"]=>
    string(6) "XXX134"
    ["bank_routing_number"]=>
    string(6) "XXXXX1"
    ["links"]=>
    array(6) {
      ["transactions"]=>
      string(64) "/disbursements/82544f72-fe75-409c-beee-77f4f23b3dda/transactions"
      ["wallet_accounts"]=>
      string(67) "/disbursements/82544f72-fe75-409c-beee-77f4f23b3dda/wallet_accounts"
      ["bank_accounts"]=>
      string(65) "/disbursements/82544f72-fe75-409c-beee-77f4f23b3dda/bank_accounts"
      ["paypal_accounts"]=>
      string(67) "/disbursements/82544f72-fe75-409c-beee-77f4f23b3dda/paypal_accounts"
      ["items"]=>
      string(57) "/disbursements/82544f72-fe75-409c-beee-77f4f23b3dda/items"
      ["users"]=>
      string(57) "/disbursements/82544f72-fe75-409c-beee-77f4f23b3dda/users"
    }
  }
  [7]=>
  array(8) {
    ["id"]=>
    string(36) "d61cb0c5-9c50-485b-8d28-10c2ed72baa6"
    ["amount"]=>
    int(100)
    ["currency"]=>
    string(3) "AUD"
    ["batch_id"]=>
    int(23984)
    ["created_at"]=>
    string(24) "2017-08-10T20:38:10.619Z"
    ["updated_at"]=>
    string(24) "2017-08-10T20:42:37.593Z"
    ["state"]=>
    string(10) "successful"
    ["links"]=>
    array(6) {
      ["transactions"]=>
      string(64) "/disbursements/d61cb0c5-9c50-485b-8d28-10c2ed72baa6/transactions"
      ["wallet_accounts"]=>
      string(67) "/disbursements/d61cb0c5-9c50-485b-8d28-10c2ed72baa6/wallet_accounts"
      ["bank_accounts"]=>
      string(65) "/disbursements/d61cb0c5-9c50-485b-8d28-10c2ed72baa6/bank_accounts"
      ["paypal_accounts"]=>
      string(67) "/disbursements/d61cb0c5-9c50-485b-8d28-10c2ed72baa6/paypal_accounts"
      ["items"]=>
      string(57) "/disbursements/d61cb0c5-9c50-485b-8d28-10c2ed72baa6/items"
      ["users"]=>
      string(57) "/disbursements/d61cb0c5-9c50-485b-8d28-10c2ed72baa6/users"
    }
  }
  [8]=>
  array(10) {
    ["id"]=>
    string(36) "4bf3b3ec-f2f2-423d-91cb-e8f419397a20"
    ["amount"]=>
    int(100)
    ["currency"]=>
    string(3) "AUD"
    ["batch_id"]=>
    int(479)
    ["created_at"]=>
    string(24) "2016-07-26T21:43:43.743Z"
    ["updated_at"]=>
    string(24) "2016-07-26T22:01:07.030Z"
    ["state"]=>
    string(7) "batched"
    ["to"]=>
    string(20) "PayPal Disbursement "
    ["paypal_email"]=>
    string(47) "f6e68777-ea86-4368-8cec-346461722206@google.com"
    ["links"]=>
    array(6) {
      ["transactions"]=>
      string(64) "/disbursements/4bf3b3ec-f2f2-423d-91cb-e8f419397a20/transactions"
      ["wallet_accounts"]=>
      string(67) "/disbursements/4bf3b3ec-f2f2-423d-91cb-e8f419397a20/wallet_accounts"
      ["bank_accounts"]=>
      string(65) "/disbursements/4bf3b3ec-f2f2-423d-91cb-e8f419397a20/bank_accounts"
      ["paypal_accounts"]=>
      string(67) "/disbursements/4bf3b3ec-f2f2-423d-91cb-e8f419397a20/paypal_accounts"
      ["items"]=>
      string(57) "/disbursements/4bf3b3ec-f2f2-423d-91cb-e8f419397a20/items"
      ["users"]=>
      string(57) "/disbursements/4bf3b3ec-f2f2-423d-91cb-e8f419397a20/users"
    }
  }
  [9]=>
  array(10) {
    ["id"]=>
    string(36) "290a6c0d-a328-465f-afd8-2ea46cc9525d"
    ["amount"]=>
    int(100)
    ["currency"]=>
    string(3) "AUD"
    ["batch_id"]=>
    int(393)
    ["created_at"]=>
    string(24) "2016-06-09T13:26:24.150Z"
    ["updated_at"]=>
    string(24) "2016-06-09T22:01:34.938Z"
    ["state"]=>
    string(7) "batched"
    ["to"]=>
    string(20) "PayPal Disbursement "
    ["paypal_email"]=>
    string(47) "914f2248-3eaf-469b-b9f0-904e36bbc4e6@google.com"
    ["links"]=>
    array(6) {
      ["transactions"]=>
      string(64) "/disbursements/290a6c0d-a328-465f-afd8-2ea46cc9525d/transactions"
      ["wallet_accounts"]=>
      string(67) "/disbursements/290a6c0d-a328-465f-afd8-2ea46cc9525d/wallet_accounts"
      ["bank_accounts"]=>
      string(65) "/disbursements/290a6c0d-a328-465f-afd8-2ea46cc9525d/bank_accounts"
      ["paypal_accounts"]=>
      string(67) "/disbursements/290a6c0d-a328-465f-afd8-2ea46cc9525d/paypal_accounts"
      ["items"]=>
      string(57) "/disbursements/290a6c0d-a328-465f-afd8-2ea46cc9525d/items"
      ["users"]=>
      string(57) "/disbursements/290a6c0d-a328-465f-afd8-2ea46cc9525d/users"
    }
  }
}
```

**Comment by jeandee** [comment]
Hello Nino,

I just checked with our lead tester and I suppose these endpoints were overlooked for the API Reference. Since they're public-facing, there's no reason for them not to be on the documentation. This looks like a task for me. Don't worry; I've added it to my backlog.

I notice that in addition to the calls already here, there is also `GET /disbursements/:id/users` and `GET disbursements/:id/items`, both taking in the id parameter. These can be added to our repository as well at low / typical priority.
